src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -1310,6 +1357,102 @@ static status_t del_routes_internal(private_kernel_super_ipsec_t* this,
 }
 
 
+METHOD(kernel_ipsec_t, write_terminator_tunnel_ike_sa_uptime, status_t,
+	private_kernel_super_ipsec_t* this,
+	ike_sa_t* ike_sa)
+{
+	(void)this;
+	const ike_super_peer_info_t* super_peer_info = ike_sa->get_super_peer_info(ike_sa);
+	if(! super_peer_info) {
+		LOG(DBG_KNL, SL_LOG_ERR, "failed to get super peer info");
+		return FAILED;
+	}
+
+	if(IKE_SUPER_PEER_INFO_TYPE_TUNNEL == super_peer_info->get_type(super_peer_info)) {
```

Comment:
These writes NEED to be for `IKE_SUPER_PEER_INFO_TYPE_TUNNEL`, not for terminator tunnels

Reply:
Note this change should also be reflected in the function name (`write_configured_tunnel_ike_sa_uptime()` and `write_configured_tunnel_ike_sa_auth_uptime()`)